command/agent/agent.go

```diff
@@ -135,75 +129,83 @@ func (c *cmd) startupJoinWan(agent *agent.Agent, cfg *config.RuntimeConfig) erro
 		return nil
 	}
 
-	c.UI.Output("Joining -wan cluster...")
+	c.logger.Info("Joining wan cluster")
 	n, err := agent.JoinWAN(cfg.StartJoinAddrsWAN)
 	if err != nil {
 		return err
 	}
 
-	c.logger.Info("Join -wan completed. Initial agents synced with", "agent_count", n)
+	c.logger.Info("Join wan completed. Initial agents synced with", "agent_count", n)
 	return nil
 }
 
 func (c *cmd) run(args []string) int {
+	ui := &mcli.PrefixedUi{
+		OutputPrefix: "==> ",
+		InfoPrefix:   "    ",
+		ErrorPrefix:  "==> ",
+		Ui:           c.ui,
+	}
+
 	if err := c.flags.Parse(args); err != nil {
 		if !strings.Contains(err.Error(), "help requested") {
-			c.UI.Error(fmt.Sprintf("error parsing flags: %v", err))
+			ui.Error(fmt.Sprintf("error parsing flags: %v", err))
 		}
 		return 1
 	}
 	if len(c.flags.Args()) > 0 {
-		c.UI.Error(fmt.Sprintf("Unexpected extra arguments: %v", c.flags.Args()))
+		ui.Error(fmt.Sprintf("Unexpected extra arguments: %v", c.flags.Args()))
 		return 1
 	}
 
-	logGate := &logging.GatedWriter{Writer: &cli.UiWriter{Ui: c.UI}}
+	// FIXME: logs should always go to stderr, but previously they were sent to
+	// stdout, so continue to use Stdout for now, and fix this in a future release.
+	logGate := &logging.GatedWriter{Writer: c.ui.Stdout()}
```

Comment:
This is the fix for the primary issue, but other changes were required to make this possible.